### PR TITLE
add  automatic generate macro script

### PR DESCRIPTION
**Pull request by helintongh**
add script to automatically generate `REFLECTION` macros.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,6 +130,80 @@ We can slove the problem1 easily with c++17:
...
+	struct person {
+		std::string name;
+		int age;
+	};
+
+	char *iguana = NULL;
+
+	struct composit_t { int a; std::vector<std::string> b; int c; std::map<int, int> d; std::unordered_map<int, int> e; double f;};
+
+	char *iguana_test = NULL;
```
Comment (qicosmos) [inline README.md:155]: I think it is unused.

File: scripts/test_macro_generator.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  std::string name;
+  one_t one;
+  int age;
+};
+
+char *iguana = NULL;
+
+struct composit_t { int a; std::vector<std::string> b; int c; std::map<int, int> d; std::unordered_map<int, int> e; double f;};
+
+char *iguana_test = NULL;
```
Comment (qicosmos) [inline scripts/test_macro_generator.cpp:21]: It's unused.

Comment (helintongh) [reply scripts/test_macro_generator.cpp:21]: Insert the normal statement for test.I will delete it.

Comment (qicosmos) [reply scripts/test_macro_generator.cpp:21]: ok，good，keep it，but comment why you add this unused code.

**Comment by qicosmos** [comment]
please rebase main

**Comment by helintongh** [comment]
rebase main done.

**Pull request merged**